src/SFA.DAS.AssessorService.Web/Controllers/CertificateHistoryController.cs

```diff
@@ -13,40 +16,110 @@
 namespace SFA.DAS.AssessorService.Web.Controllers
 {
     [PrivilegeAuthorize(Privileges.ViewCompletedAssessments)]
-    [CheckSession]
-    [Route("[controller]/[action]")]
+    [CheckSession(CertificateHistoryRoute, nameof(ResetSession), nameof(ICertificateHistorySession.CertificateHistorySearchTerm))]
+    [Route(CertificateHistoryRoute)]
     public class CertificateHistoryController : Controller
     {
         private readonly ILogger<CertificateController> _logger;
         private readonly IHttpContextAccessor _contextAccessor;
-        private readonly ICertificateApiClient _certificateApiClient;        
-        private readonly ISessionService _sessionService;
-
+        private readonly ICertificateApiClient _certificateApiClient;
+        private readonly ICertificateHistorySession _certificateHistorySession;
+        private const string CertificateHistoryRoute = "certificateHistory";
+        private const int DefaultPageIndex = 1;
         public CertificateHistoryController(
             ILogger<CertificateController> logger,
             IHttpContextAccessor contextAccessor,
-            ICertificateApiClient certificateApiClient,        
-            ISessionService sessionService)
+            ICertificateApiClient certificateApiClient,
+            ICertificateHistorySession certificateHistorySession)
         {
             _logger = logger;
             _contextAccessor = contextAccessor;
             _certificateApiClient = certificateApiClient;
-            _sessionService = sessionService;       
+            _certificateHistorySession = certificateHistorySession;
+        }
+
+        public async Task<IActionResult> Index()
+        {
+            var vm = await MapViewModelFromSession();
+            return View(nameof(Index), vm);
+        }
+
+        [HttpGet(nameof(ResetSession))]
+        [CheckSession(nameof(ICertificateHistorySession.CertificateHistorySearchTerm), CheckSession.Ignore)]
+        public IActionResult ResetSession()
+        {
+            SetDefaultSession();
+            return RedirectToAction(nameof(Index));
+        }
+
+        [HttpGet(nameof(Search))]
+        public async Task<IActionResult> Search(string searchTerm)
+        {
+            _certificateHistorySession.CertificateHistorySearchTerm = searchTerm ?? string.Empty;
+
+            // reset the page indexes as the new results may have less pages
+            _certificateHistorySession.CertificateHistoryPageIndex = 1;
```

Comment:
There is a DefaultPageIndex you could use

Reply:
Changed